sdk/template/template/test/public/core-v1-test.spec.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+import { isPlaybackMode } from "@azure-tools/test-recorder";
+import { Context } from "mocha";
+import { QueueServiceClient } from "@azure/storage-queue";
+import {
+  TestProxyHttpClientCoreV1,
+  env,
+  RecorderStartOptions,
+  SanitizerOptions
+} from "@azure-tools/test-recorder-new";
+
+// This fake URL will be used in the recording. We use a general regex sanitizer
+// (below) to replace the real SAS URL from when we record the test with this
+// fake version.
+const fakeSASUrl =
+  "https://account_name.queue.core.windows.net/?sv=2020-08-04&ss=bfqt&srt=sco&sp=rwdlacuptfx&se=2026-07-10T07:00:24Z&st=2021-07-09T23:00:24Z&spr=https&sig=fake_sig";
+
+function getRecorderStartOptions(): RecorderStartOptions {
+  // These options are passed to the recorder for playback.
+  const envSetupForPlayback: Record<string, string> = {
+    // In playback, we use the fake SAS URL instead of a real one which would have
+    // been provided via the .env file or otherwise.
+    STORAGE_SAS_URL: fakeSASUrl
+  };
+
+  let sanitizerOptions: SanitizerOptions | undefined;
+
+  if (env.STORAGE_SAS_URL) {
+    const realSASParts = env.STORAGE_SAS_URL.split("/");
+    const fakeSASParts = fakeSASUrl.split("/");
+
+    sanitizerOptions = {
+      // There are several different sanitzers that can be used. A general regex sanitizer
+      // allows for a simple find and replace based on a given regular expression. This is
+      // used here to strip out the sensitive and dynamic parts of the SAS URL in the recording,
+      // replacing them with their corresponding components from the fake SAS URL defined above.
+      generalRegexSanitizers: [
+        // This one replaces the host portion of the URL, which is dynamic based on the storage account...
+        {
+          regex: realSASParts[2],
```

Comment:
So these are regex programs as strings, and not actual RegExp instances? Which regexp syntax are we expected to use to configure these? Are they interpreted as JavaScript regexp or are they compiled by the proxy tool?

Reply:
They're compiled by the proxy tool, so are .NET regular expressions. In practice I believe the syntax is very similar, if not a superset of, the JS regexp syntax.

Reply:
Here's a comparison I found: https://www.regular-expressions.info/dotnet.html#ecmascript

Looks like there are some differences.